iroh/src/magicsock.rs

```diff
@@ -2768,18 +2904,41 @@ impl QuicMappedAddr {
         addr[1..6].copy_from_slice(&Self::ADDR_GLOBAL_ID);
         addr[6..8].copy_from_slice(&Self::ADDR_SUBNET);
 
-        let counter = ADDR_COUNTER.fetch_add(1, Ordering::Relaxed);
+        let counter = NODE_ID_ADDR_COUNTER.fetch_add(1, Ordering::Relaxed);
         addr[8..16].copy_from_slice(&counter.to_be_bytes());
 
-        Self(SocketAddr::new(IpAddr::V6(Ipv6Addr::from(addr)), 12345))
+        Self(SocketAddr::new(
+            IpAddr::V6(Ipv6Addr::from(addr)),
+            MAPPED_ADDR_PORT,
+        ))
     }
 }
 
-impl std::fmt::Display for QuicMappedAddr {
+impl TryFrom<SocketAddr> for NodeIdMappedAddr {
+    type Error = anyhow::Error;
+
+    fn try_from(value: SocketAddr) -> std::result::Result<Self, Self::Error> {
+        match value {
+            SocketAddr::V4(_) => anyhow::bail!("NodeIdMappedAddrs are all Ipv6, addr {value:?}"),
+            SocketAddr::V6(addr) => {
+                if addr.port() != MAPPED_ADDR_PORT {
+                    anyhow::bail!("not a mapped addr");
+                }
+                let octets = addr.ip().octets();
+                if octets[6..8] != NodeIdMappedAddr::ADDR_SUBNET {
```

Comment:
Likewise this should check the entire prefix I think.  Yes, previously this allowed just any IP address and hoped for the best.  But now we have two types that's no longer an option so we may as well do the matching slightly better.  The code in the magicsock now has to deal with neither matching already anyway.